app/src/main/java/com/example/drawingApp/DrawingViewModel.kt

```diff
@@ -15,8 +15,16 @@ class DrawingViewModel : ViewModel() {
 
     var activeBitmap: Bitmap? = null
 
-    var isLayerCreationDialogOpen = false
-    var currentLayerCreationDialogText = ""
+    var isLayerCreationDialogOpen = false   //new Layer
+    var currentLayerDialogText = ""
```

Comment:
I like nullable declarations, I think working around the null makes your code safer in case anything in future were to change something and it went wrong, you would have robust code to handle the null in a way that doesn't crash the app.

But if we're honest here, the logic in this app won't be touched by others and its likely safe to include variables like this.

I hope.

Reply:
and again the empty string acts as the effective null

Reply:
I don't think it's good practice to think about if the code will be touched by others. We want to make it good code regardless so we grow into good practices.
I think this field would be better served as nullable, but you can set it up either way obviously. Empty is fine too